muspinsim/cpp/parallel.cpp

```diff
@@ -0,0 +1,239 @@
+#include "base.h"
+
+/*
+ * Computes V^\dagger (M \otimes 1_{d}) V between a complex vector V and a
+ * matrix M where 1_{d} an identity matrix of size d
+ *
+ * For speed, no checks are performed here. The matrix M should be Hermitian
+ * with size N and the shape of V should be (N*d, 1).
+ *
+ * @param V The vector
+ * @param M The matrix
+ * @param d The dimension of the identity matrix
+ *
+ * @return The result of the combined product. Will be a single value.
+ */
+double parallel::fast_measure(np_array_complex_t& V, np_array_complex_t& M, long int d) {
```

Comment:
Unless I'm missing something, we only seem to actually call `fast_evolve_ptr` and `fast_measure_h_ptr`, the non-hermitian measure and all of the "python" methods are unused? Is this like `_fast_evolve_python` where there is currently no way of accessing it, but we want to have it available as a library function in case there's a user case in the future?

Reply:
Yes it was the first method I made, I found it useful when testing and I believe there may be a case for it elsewhere.